chore: Add filters and words so full repo passes cSpell

nschonni opened the pull request:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

nschonni committed:
chore: Move extra words back to custom-words.txt

nschonni committed:
typo: multilanguage -> multilingual

nschonni committed:
typo: supress -> suppress

nschonni committed:
typo: Supressions -> Suppressions

nschonni committed:
typo: choosed -> chosen

nschonni committed:
typo: Asssignment -> Assignment

nschonni committed:
typo: doesnot -> does not

nschonni committed:
typo: proeprties -> properties

nschonni committed:
typo: Listby -> ListBy

nschonni committed:
typo: statuscode -> statusCode

nschonni committed:
typo: outputm -> output

nschonni committed:
typo: Dataplane -> Data plane

nschonni committed:
typo: Powershell's -> PowerShell's

nschonni committed:
typo: schemaregistry -> schema registry

nschonni committed:
typo: extendedlocation -> extendedLocation

nschonni committed:
typo: pagable -> pageable

nschonni committed:
typo: flim -> film

nschonni committed:
typo: reponse -> response

nschonni committed:
chore: Add filters and words so full repo passes

AutorestCI commented:
You don't have permission to trigger SDK Automation.
Please add yourself to ***Azure*** group from opensource portal if you are MSFT employee,
or please ask reviewer to add comment *** /openapibot sdkautomation ***.
Please ask tih@microsoft.com (or NullMDR in github) for additional help.

nschonni commented:
I didn't update the CI workflow in script/spellcheck.ts because the filtering from the util is also used by the prettier task and would cause a bunch of other formatting changes

heaths reviewed (approved):
Search and Key Vault changes LGTM. I didn't review the others.

nschonni committed:
typo: subcription -> subscription

nschonni committed:
typo: connot -> cannot

nschonni committed:
typo: systemdata -> systemData

nschonni committed:
typo: exsisting -> existing

nschonni committed:
typo: commmand -> command

nschonni committed:
typo: paramter -> parameter

nschonni commented:
I've cherry-picked a few of the "functional" pieces in here to separate PRs since this larger change seems stalled

nschonni committed:
chore: Bump cSpell to v5

Removes some misspellings in the dictionaries and speeds up execution significantly

nschonni committed:
chore: Allow compound words in Markdown

nschonni committed:
chore: Sort and flatten custom-words.txt

nschonni committed:
typo: additinal -> additional

nschonni committed:
typo: syncronous -> synchronous
